### PR TITLE
[image][android] Add resolveAssetSource method

### PR DESCRIPTION
**Pull request by mstach60161**
# Why

To extend the functionality of `expo-image`.

# How

- Import `resolveAssetSource` from ReactNative image.
- Wrap it in expo-image `resolveAssetSource`.

# Test Plan

- Pass the source to `resolveAssetSource` method and check if it's properly resolved.

# Checklist

- [x] Documentation is up to date to reflect these changes (eg: https://docs.expo.io and README.md).

### CODE REVIEW

File: packages/expo-image/src/Image.tsx [hunk 1]
```diff
@@ -79,6 +80,17 @@ export default class Image extends React.Component<ImageProps, ImageState> {
...
   }
 
+  /**
+   * Resolves an asset reference into an object which has the properties `uri`, `width` and `height`
+   *
+   * @param source A number (opaque type returned by require('./foo.png')) or an `ImageSource`.
+   *
+   * @return an object constaining `uri` `width` and `height`.
+   */
+  static resolveAssetSource(source: ImageSourcePropType): object {
```
Comment (barthap) [inline packages/expo-image/src/Image.tsx:90]:
```suggestion
  static resolveAssetSource(source: ImageSourcePropType): ImageAssetSource {
```
Please define somehow the return type, RN image defines it as follows: _(copied from `react-native/index.d.ts`)_
```ts
/**
 * @see https://reactnative.dev/docs/image#resolveassetsource
 */
export interface ImageResolvedAssetSource {
    height: number;
    width: number;
    scale: number;
    uri: string;
}
```

**Review by barthap — CHANGES_REQUESTED** [review]

File: packages/expo-image/src/Image.types.ts [hunk 2]
```diff
@@ -31,6 +31,13 @@ interface AndroidGlideException extends AndroidThrowable {
   causes: AndroidThrowable[] | null;
 }
 
+export interface ImageResolvedAssetSource {
+  height: number;
+  width: number;
+  scale: number;
+  uri: string;
+}
+
```
Comment (kkafar) [inline packages/expo-image/src/Image.types.ts:40]:
```suggestion
```

It is fine this way, however in situation when we are exporting RN method I would consider also exporting RN interface (as long as it is exact copy of it). Up to you :smile:

**Review by kkafar — COMMENTED** [review]
Looks good :smile:

File: packages/expo-image/src/Image.types.ts (hunk 2 above)
Comment (mstach60161) [reply packages/expo-image/src/Image.types.ts:40]: I also considered it, but I can't decide which solution is better. 😅😅

**Pull request closed without merging**